src/openforms/registrations/contrib/zgw_apis/tests/test_migrations.py

```diff
@@ -201,3 +201,150 @@ def test_no_zgw_api_group_created(self):
+            "drc_service": drc,
+            "ztc_service": ztc,
+        }
+
+        # create different config groups & default config
+        group1 = ZGWApiGroupConfig.objects.create(
+            zaaktype="http://www.example-ztc.com/api/v1/zaaktypen/1",
+            informatieobjecttype="http://www.example-ztc.com/api/v1/informatieobjecttypen/1",
+            **services,
+        )
+        group2 = ZGWApiGroupConfig.objects.create(
+            zaaktype="http://www.example-ztc.com/api/v1/zaaktypen/2",
+            informatieobjecttype="http://www.example-ztc.com/api/v1/informatieobjecttypen/2",
+            **services,
+        )
+        ZgwConfig.objects.create(default_zgw_api_group=group1)
+
+        # form with different backend options
+        form = Form.objects.create(name="test form")
+        self.backend1 = FormRegistrationBackend.objects.create(
+            form=form,
+            key="global-defaults",
+            name="global-defaults",
+            backend="zgw-create-zaak",
+            options={},
+        )
+        self.backend2 = FormRegistrationBackend.objects.create(
+            form=form,
+            key="different-group",
+            name="different-group",
+            backend="zgw-create-zaak",
+            options={"zgw_api_group": group2.id},
+        )
+        self.backend3 = FormRegistrationBackend.objects.create(
+            form=form,
+            key="explicit-types-set",
+            name="explicit-types-set",
+            backend="zgw-create-zaak",
+            options={
+                "zaaktype": "http://www.example-ztc.com/api/v1/zaaktypen/3",
+                "informatieobjecttype": "http://www.example-ztc.com/api/v1/informatieobjecttypen/3",
+            },
+        )
+        self.backend4 = FormRegistrationBackend.objects.create(
+            form=form,
+            key="explicit-types-set-explicit-group",
+            name="explicit-types-set-explicit-group",
+            backend="zgw-create-zaak",
+            options={
+                "zgw_api_group": group2.id,
+                "zaaktype": "http://www.example-ztc.com/api/v1/zaaktypen/3",
+                "informatieobjecttype": "http://www.example-ztc.com/api/v1/informatieobjecttypen/3",
+            },
+        )
+
+    def test_defaults_added_to_backends(self):
+        FormRegistrationBackend = self.apps.get_model(
+            "forms", "FormRegistrationBackend"
+        )
+        backend1 = FormRegistrationBackend.objects.get(id=self.backend1.id)
+        backend2 = FormRegistrationBackend.objects.get(id=self.backend2.id)
+        backend3 = FormRegistrationBackend.objects.get(id=self.backend3.id)
+        backend4 = FormRegistrationBackend.objects.get(id=self.backend4.id)
+
+        with self.subTest("backend 1"):
+            options = backend1.options
+            self.assertIn("zaaktype", options)
+            self.assertEqual(
+                options["zaaktype"], "http://www.example-ztc.com/api/v1/zaaktypen/1"
```

Comment:
Are the `assertIn` checks redundant? If they are not present, we will get a `KeyError` anyway?

Reply:
yeah, but the assert in makes sure you get a Failure rather than an Error in the output :) makes it explicit that it's not a bug in the test itself